sharedbuffers/mapped_struct.py

```diff
@@ -1110,24 +1109,24 @@ def __contains__(self, elem):
             except (ValueError, TypeError):
                 return False
 
+        lo = 0
+        hi = len(self.objlist)
+
         if cython.compiled:
             dcode, _, _, offset, _ = self.objlist._metadata()
             if dcode in ('q', 'I', 'i', 'd'):
-                i1 = self._contains(elem, dcode, offset)
-                return self.objlist._getitem(i1) == elem
+                lo = self._search_key(elem, dcode, offset, lo, hi, hi // 2)
```

Comment:
If we're sorting by stable hash, keys are hashes. As such, you can compute a better hint, by using the same hinting logic StringIdMappers use.

Reply:
But this code is only called when the underlying proxied list is numeric, so the keys aren't the stable hashes. For those cases, there's the basic binary search implemented below.

Reply:
Oh, yes. Sorry, brainfart.